Octokit/Clients/MergingClient.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System.Threading.Tasks;
+
+namespace Octokit
+{
+    /// <summary>
+    /// A client for GitHub's Git Merging API.
+    /// </summary>
+    /// <remarks>
+    /// See the <a href="https://developer.github.com/v3/repos/merging/">Git Merging API documentation</a> for more information.
+    /// </remarks>
+    public class MergingClient : ApiClient, IMergingClient
+    {
+        public MergingClient(IApiConnection apiConnection) : base(apiConnection)
+        {
+        }
+
+        /// <summary>
+        /// Create a merge for a given repository
+        /// </summary>
+        /// <remarks>
+        /// http://developer.github.com/v3/repos/merging/#perform-a-merge
+        /// </remarks>
+        /// <param name="owner">The owner of the repository</param>
+        /// <param name="name">The name of the repository</param>
+        /// <param name="merge">The merge to create</param>
+        /// <returns></returns>
+        public Task<Merge> Create(string owner, string name, NewMerge merge)
+        {
+            Ensure.ArgumentNotNullOrEmptyString(owner, "owner");
+            Ensure.ArgumentNotNullOrEmptyString(name, "name");
+            Ensure.ArgumentNotNull(merge, "merge");
+            Ensure.ArgumentNotNullOrEmptyString(merge.Base, "baseBranch");
+            Ensure.ArgumentNotNullOrEmptyString(merge.Head, "head");
```

Comment:
These last two `Ensure.ArgumentNotNullOrEmptyString` calls should be in the constructor of `NewMerge` not here. We reserve argument null exceptions for when the argument is null, not when a property of the argument is null.